cranelift-codegen/src/legalizer/mod.rs

```diff
@@ -149,29 +148,43 @@ fn expand_cond_trap(
     // Split the EBB after `inst`:
     //
     //     trapnz arg
+    //     ..
     //
     // Becomes:
     //
-    //     brz arg, new_ebb
+    //     brz arg, new_ebb_resume
+    //     jump new_ebb_trap
+    //
+    //   new_ebb_trap:
     //     trap
-    //   new_ebb:
     //
+    //   new_ebb_resume:
+    //     ..
     let old_ebb = func.layout.pp_ebb(inst);
-    let new_ebb = func.dfg.make_ebb();
+    let new_ebb_trap = func.dfg.make_ebb();
+    let new_ebb_resume = func.dfg.make_ebb();
+
```

Comment:
Should be able to use the ebb splitting method to create new_ebb_trap?  I confess I'm not yet an expert on how this code all works, so I'm not 100% sure, but it does seem so.

Reply:
If there is an ebb splitting method, I do not know it yet. Where should I look for it?

Reply:
Looking at the code, it seems simpler and more coherent to use `insert_ebb` as the surrounding code is adding instruction. `split_ebb` sounds more adapted to breaking an existing block in 2, without changing any of the instruction.

Maybe I am wrong …